contrib/platform/src/com/sun/jna/platform/win32/ShellAPI.java

```diff
@@ -209,4 +211,853 @@ protected List getFieldOrder() {
+		 * frequently used programs.
+		 * </p>
+		 * </dd>
+		 * <dt>
+		 * <p>
+		 * <strong>SEE_MASK_FLAG_HINST_IS_SITE</strong> (0x08000000)
+		 * </p>
+		 * </dt>
+		 * <dd>
+		 * <p>
+		 * <strong>Introduced in Windows&nbsp;8</strong>. The
+		 * <strong>hInstApp</strong> member is used to specify the of an object
+		 * that implements . This object will be used as a site pointer. The
+		 * site pointer is used to provide services to the function, the handler
+		 * binding process, and invoked verb handlers.
+		 * </p>
+		 * </dd>
+		 * </dl>
+		 */
+		public int fMask;
+	
+		/**
+		 * <p>
+		 * Type: <strong>HWND</strong>
+		 * </p>
+		 * <p>
+		 * Optional. A handle to the parent window, used to display any message
+		 * boxes that the system might produce while executing this function.
+		 * This value can be <strong>NULL</strong>.
+		 * </p>
+		 */
+		public HWND hwnd;
+	
+		/**
+		 * <p>
+		 * Type: <strong>LPCTSTR</strong>
+		 * </p>
+		 * </dd>
+		 * <dd>
+		 * <p>
+		 * A string, referred to as a <em>verb</em>, that specifies the action
+		 * to be performed. The set of available verbs depends on the particular
+		 * file or folder. Generally, the actions available from an object's
+		 * shortcut menu are available verbs. This parameter can be
+		 * <strong>NULL</strong>, in which case the default verb is used if
+		 * available. If not, the "open" verb is used. If neither verb is
+		 * available, the system uses the first verb listed in the registry. The
+		 * following verbs are commonly used:
+		 * </p>
+		 * <dl class="indent">
+		 * <dt>
+		 * <p>
+		 * <strong>edit</strong>
+		 * </p>
+		 * </dt>
+		 * <dd>
+		 * <p>
+		 * Launches an editor and opens the document for editing. If
+		 * <strong>lpFile</strong> is not a document file, the function will
+		 * fail.
+		 * </p>
+		 * </dd>
+		 * <dt>
+		 * <p>
+		 * <strong>explore</strong>
+		 * </p>
+		 * </dt>
+		 * <dd>
+		 * <p>
+		 * Explores the folder specified by <strong>lpFile</strong>.
+		 * </p>
+		 * </dd>
+		 * <dt>
+		 * <p>
+		 * <strong>find</strong>
+		 * </p>
+		 * </dt>
+		 * <dd>
+		 * <p>
+		 * Initiates a search starting from the specified directory.
+		 * </p>
+		 * </dd>
+		 * <dt>
+		 * <p>
+		 * <strong>open</strong>
+		 * </p>
+		 * </dt>
+		 * <dd>
+		 * <p>
+		 * Opens the file specified by the <strong>lpFile</strong> parameter.
+		 * The file can be an executable file, a document file, or a folder.
+		 * </p>
+		 * </dd>
+		 * <dt>
+		 * <p>
+		 * <strong>print</strong>
+		 * </p>
+		 * </dt>
+		 * <dd>
+		 * <p>
+		 * Prints the document file specified by <strong>lpFile</strong>. If
+		 * <strong>lpFile</strong> is not a document file, the function will
+		 * fail.
+		 * </p>
+		 * </dd>
+		 * <dt>
+		 * <p>
+		 * <strong>properties</strong>
+		 * </p>
+		 * </dt>
+		 * <dd>
+		 * <p>
+		 * Displays the file or folder's properties.
+		 * </p>
+		 * </dd>
+		 * </dl>
+		 */
+		public WString lpVerb;
```

Comment:
The default W32API options handle automatically both the -W suffix and wide string -> String mapping.

So drop any -W suffixes unless the call has an explicit native wchar_t reference and use String throughout (in some uncommon instances the function will use a width-specific structure which requires both "ansi” and “unicode” variants).

> On Nov 9, 2015, at 1:43 AM, Lyor Goldstein <notifications@github.com> wrote:
> 
> In contrib/platform/src/com/sun/jna/platform/win32/ShellAPI.java:
> 
> > +		 * fail.
> > +		 * </p>
> > +		 * </dd>
> > +		 * <dt>
> > +		 * <p>
> > +		 * <strong>properties</strong>
> > +		 * </p>
> > +		 * </dt>
> > +		 * <dd>
> > +		 * <p>
> > +		 * Displays the file or folder's properties.
> > +		 * </p>
> > +		 * </dd>
> > +		 * </dl>
> > +		 */
> > +		public WString lpVerb;
> 
> Try using Java's String instead of WString - I believe JNA's code knows how to handle the conversion (make sure via a unit test)
> 
> —
> Reply to this email directly or view it on GitHub.
>